hkmc2/shared/src/test/mlscript/basics/Inheritance.mls

```diff
@@ -89,3 +89,27 @@ data class Some[out T](x: T) extends Option[T]
 //│ = [true, true, false, false]
 
 
+:todo
+data class Barr(x: Int)(y: Int) with
+  fun f = x + y
+
+:todo
+data class Bazz(z: Int) extends Bar(z % 10)(z / 10) with
+  fun g = z
+//│ ╔══[COMPILATION ERROR] Inheriting class with multiple parameter lists is not supported
+//│ ║  l.97: 	data class Bazz(z: Int) extends Bar(z % 10)(z / 10) with
+//│ ╙──      	                                ^^^^^^^^^^^^^^^^^^^
+
+:todo
+Bazz(42).f
+//│ ═══[RUNTIME ERROR] Error: Access to required field 'f' yielded 'undefined'
+
+:todo
+new Barr(40)(2) with
+  fun f = 43
+//│ ╔══[COMPILATION ERROR] Inheriting class with multiple parameter lists is not supported
+//│ ║  l.108: 	new Barr(40)(2) with
+//│ ╙──       	    ^^^^
+//│ = [function]
```

Comment:
I found that the current implementation for curried classes does not support inheritance (I explicitly reject it for now and added some cases for tracking).